mms/model_service/model_service.py

```diff
@@ -94,26 +94,33 @@ def inference(self, data):
             data to be sent back
         '''
         pre_start_time = time.time()
+
         data = self._preprocess(data)
-        infer_start_time = time.time()
 
         # Update preprocess latency metric
+        infer_start_time = time.time()
         pre_time_in_ms = (infer_start_time - pre_start_time) * 1000
         if self.model_name + '_LatencyPreprocess' in MetricsManager.metrics:
             MetricsManager.metrics[self.model_name + '_LatencyPreprocess'].update(pre_time_in_ms)
 
         data = self._inference(data)
-        data = self._postprocess(data)
 
+        post_start_ms = time.time()
         # Update inference latency metric
-        infer_time_in_ms = (time.time() - infer_start_time) * 1000
+        infer_time_in_ms = (post_start_ms - infer_start_time) * 1000
```

Comment:
Did you verify this change? MXNet is doing lazy evaluation, so you only get the inference completed when you extract the output. Please verify this closely to make sure the metrics you have are correct.